cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -324,88 +346,270 @@ fn constant_true() -> bool {
-        let context = parse_context(self.context, schema.as_ref(), &action)?;
-        let q = Request::new(
+        let mut errs = vec![];
+        let mut warnings = vec![];
+        let schema = match self.schema.map(Schema::parse).transpose() {
+            Ok(None) => None,
+            Ok(Some((schema, new_warnings))) => {
+                warnings.extend(new_warnings);
+                Some(schema)
+            }
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let principal = match self
+            .principal
+            .map(|p| parse_entity_uid(p, "principal"))
+            .transpose()
+        {
+            Ok(None) => None,
+            Ok(Some(euid)) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let action = match parse_action(self.action) {
+            Ok(euid) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let resource = match self
+            .resource
+            .map(|r| parse_entity_uid(r, "resource"))
+            .transpose()
+        {
+            Ok(None) => None,
+            Ok(Some(euid)) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let context = match parse_context(self.context, schema.as_ref(), action.as_ref()) {
+            Ok(context) => Some(context),
+            Err(e) => {
+                errs.extend(e);
+                None
+            }
+        };
+        let q = match Request::new(
             principal,
-            Some(action),
+            action,
             resource,
-            context,
+            context.unwrap_or_else(|| Context::empty()),
             if self.enable_request_validation {
                 schema.as_ref()
             } else {
                 None
             },
         )
-        .map_err(|e| [e.to_string()])?;
-        let (policies, entities) = self.slice.try_into(schema.as_ref())?;
-        Ok((q, policies, entities))
+        .map_err(|e| e.to_string())
+        {
+            Ok(req) => Some(req),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let (policies, entities) = match self.slice.try_into(schema.as_ref()) {
+            Ok((policies, entities)) => (Some(policies), Some(entities)),
+            Err(e) => {
+                errs.extend(e);
+                (None, None)
+            }
+        };
+        if errs.is_empty() {
+            // TODO: this ignores warnings
+            let Some(q) = q else {
+                return Err(errs);
+            };
+            let Some(policies) = policies else {
+                return Err(errs);
+            };
+            let Some(entities) = entities else {
+                return Err(errs);
+            };
```

Comment:
These cases shouldn't be possible with an empty `errs`, right?

Reply:
Correct.  But I thought returning `Err([])` in this impossible case was better than panicking

Reply:
It still feels a little ugly to me 🙂 would prefer if you could refactor to remove, but it's not a blocking issue.

Reply:
Refactored